storage/backup/backup-cli/src/metadata/view.rs

```diff
@@ -4,38 +4,41 @@
 
 use crate::{
     metadata::{
-        EpochEndingBackupMeta, IdentityMeta, Metadata, StateSnapshotBackupMeta,
-        TransactionBackupMeta,
+        CompactionTimestampsMeta, EpochEndingBackupMeta, IdentityMeta, Metadata,
+        StateSnapshotBackupMeta, TransactionBackupMeta,
     },
-    storage::FileHandle,
+    storage::{FileHandle, TextLine},
 };
 use anyhow::{anyhow, ensure, Result};
 use aptos_types::transaction::Version;
 use itertools::Itertools;
-use std::{fmt, str::FromStr};
+use std::{collections::HashMap, fmt, str::FromStr};
 
-#[derive(PartialEq, Debug)]
+#[derive(Debug)]
 pub struct MetadataView {
     epoch_ending_backups: Vec<EpochEndingBackupMeta>,
     state_snapshot_backups: Vec<StateSnapshotBackupMeta>,
     transaction_backups: Vec<TransactionBackupMeta>,
     _identity: Option<IdentityMeta>,
-    file_handles: Option<Vec<FileHandle>>,
+    // The compaction timestamp of the file handles producing this view
+    compaction_timestamps: Option<CompactionTimestampsMeta>,
 }
 
 impl MetadataView {
-    pub(crate) fn new(metadata_vec: Vec<Metadata>, file_handles: Vec<FileHandle>) -> Self {
+    pub(crate) fn new(metadata_vec: Vec<Metadata>) -> Self {
         let mut epoch_ending_backups = Vec::new();
         let mut state_snapshot_backups = Vec::new();
         let mut transaction_backups = Vec::new();
         let mut identity = None;
+        let mut compaction_timestamps = Option::None;
 
         for meta in metadata_vec {
             match meta {
                 Metadata::EpochEndingBackup(e) => epoch_ending_backups.push(e),
                 Metadata::StateSnapshotBackup(s) => state_snapshot_backups.push(s),
                 Metadata::TransactionBackup(t) => transaction_backups.push(t),
                 Metadata::Identity(i) => identity = Some(i),
+                Metadata::CompactionTimestamps(t) => compaction_timestamps = Some(t),
```

Comment:
Feels more straightforward to keep multiple of them and sort them by timestamp as we talked about yesterday? 

```
pub struct CompactionTimestampMeta {
    latest_compaction_timestamp: u64,
    compaction_timestamps: HashMap<FileHandle, u64>,
}

```

(And do we actually want to keep the new filenames that don't have a timestamp in this file?)